Utilities/macrecovery/macrecovery.py

```diff
@@ -7,28 +7,30 @@
 """
 
 import argparse
-import binascii
 import hashlib
 import json
 import linecache
 import os
 import random
 import struct
+import string
 import sys
 
 try:
     from urllib.request import Request, HTTPError, urlopen
     from urllib.parse import urlparse
 except ImportError:
-    from urllib2 import Request, HTTPError, urlopen
-    from urlparse import urlparse
+    print('ERROR: Python 2 is not supported, please use Python 3')
+    sys.exit(1)
 
 SELF_DIR = os.path.dirname(os.path.realpath(__file__))
+TERMINAL_SIZE = os.get_terminal_size().columns
```

Comment:
Thanks for progress bar work. Imo it looks good. But it does not handle well, a) resizing screen while downloading, nor b) screen too narrow.
Should probably refetch terminal size every time it needs it, should also revert to old method below a certain terminal width (min. width at which progress bar works).